src/DeathWindow.cpp

```diff
@@ -1,96 +1,128 @@
+#include "DeathWindow.h"
+
 #include <Application.h>
-#include <Window.h>
-#include <ListView.h>
-#include <ScrollView.h>
-#include <CheckBox.h>
-#include <Roster.h>
+#include <Beep.h>
+#include <LayoutBuilder.h>
 #include <Messenger.h>
 #include <OS.h>
-#include <Beep.h>
+#include <Roster.h>
 
-#include "DeathWindow.h"
 
+DeathWindow::DeathWindow()
+	:
+	BWindow(BRect(), "BeDead", B_TITLED_WINDOW,
+			 B_AUTO_UPDATE_SIZE_LIMITS | B_WILL_ACCEPT_FIRST_CLICK)
+{
+	ResizeTo(300, 280);
+	CenterOnScreen();
 
-const uint32 KILL_KILL_KILL = 'KiLL';
-int32 PIDIndex[10000];
+	fBackView = new BView("BackView", B_WILL_DRAW);
+	rgb_color bgcolor = ui_color(B_PANEL_BACKGROUND_COLOR);
+	fBackView->SetViewColor(bgcolor);
 
-DeathWindow::DeathWindow() : BWindow(BRect(100,100,400,400),"BeDead",B_TITLED_WINDOW,B_NOT_RESIZABLE | B_NOT_ZOOMABLE | B_WILL_ACCEPT_FIRST_CLICK) {
+	fDeathList = new BListView("Death List", B_SINGLE_SELECTION_LIST,
+					B_WILL_DRAW);
 
-	DeathList = new BListView(BRect(0,0,285,250),"Death List",B_SINGLE_SELECTION_LIST);
+	fDeathScroll = new BScrollView("scroll_death", fDeathList,							B_WILL_DRAW, false, true);
+	fDeathScroll->SetExplicitMinSize(BSize(260, 260));
 
-	BScrollView *DeathScroll=new BScrollView("scroll_death",DeathList,B_FOLLOW_ALL_SIDES,0,false,true);	
-	
-	BMessage *KiLL=new BMessage(KILL_KILL_KILL);
-	DeathList->SetInvocationMessage(KiLL);
+	fDeathList->SetInvocationMessage(new BMessage(kAppKill));
 
-	CloseCheck=new BCheckBox(BRect(15,260,200,280),"CloseCheck","Close on Kill",new BMessage(B_OK));
-	CloseCheck->SetValue(B_CONTROL_ON);
+	fCloseCheck = new BCheckBox("CloseCheck", "Close BeDead on Kill", NULL, NULL);
+	fCloseCheck->SetValue(B_CONTROL_OFF);
 		
-	DeathList->AttachedToWindow();
-	CloseCheck->AttachedToWindow();
-
-	AddChild(DeathScroll);
-	AddChild(CloseCheck);
-
-	AddApps(DeathList);
+	fDeathList->AttachedToWindow();
+	fCloseCheck->AttachedToWindow();
+
+	BLayoutBuilder::Group<>(fBackView, B_VERTICAL)
+		.Add(fDeathScroll)
+		.AddGroup(B_HORIZONTAL)
+			.Add(fCloseCheck)
+			.AddGlue()
+		.End()
+		.End();
+
+	BLayoutBuilder::Group<>(this, B_VERTICAL, 0)
+		.AddGroup(B_VERTICAL)
+			.SetInsets(B_USE_WINDOW_SPACING)
+			.Add(fBackView)
+			.End();
+
+	AddApps(fDeathList);
 	
 	BMessenger mess(this);
 	be_roster->StartWatching(mess);
 	
-	Show();
 }
 
-DeathWindow::~DeathWindow() {
+
+DeathWindow::~DeathWindow()
+{
 	BMessenger mess(this);
 	be_roster->StopWatching(mess);
 }
 
-void DeathWindow::MessageReceived(BMessage *message) {
+
+void
+DeathWindow::MessageReceived(BMessage *message)
+{
 	switch(message->what) {
 		case B_SOME_APP_LAUNCHED:
-			DeathList->MakeEmpty();
-			AddApps(DeathList);		
+			fDeathList->MakeEmpty();
+			AddApps(fDeathList);
 			break;
+
```

Comment:
cases need extra { } as there are more than one instructions in the case blocks.